src/test/ci/test_tested_versions.py

```diff
@@ -0,0 +1,150 @@
+from ci.tested_versions_utils import (
+    NonSemanticVersion,
+    SemanticVersion,
+    TestedVersions,
+    parse_version,
+    _get_supported_version_ranges,
+)
+
+
+def test_parse_version():
+    assert parse_version("  1.2.3  # test!") == SemanticVersion(
+        True, "1.2.3  # test!", 1, 2, 3, "", "test!"
+    )
+
+    # We disregard comments in the equality check
+    assert parse_version("!1.2.3") == SemanticVersion(
+        False, "1.2.3  # test!", 1, 2, 3, "", "test!"
+    )
```

Comment:
Usually we put each block like this in a different test and the comment inside is the name of the test. You can use `pytest.mark.parametrize` to make it nicer.
Example:
![image](https://user-images.githubusercontent.com/49228804/180210903-183573b9-4114-4b82-ab85-7b60c6d961e7.png)

Reply:
Yeah, but do you want to compare the understandably of the current status and using a parametrization to.... what? save you from typing `assert`? It may look cooler, but understandability goes down a hole so deep that you eventually find yourself in China.

Reply:
Tests split in b4df02d864879cc357328295b6d332c45ba553b5